websites/R/Reaper Scans/presence.ts

```diff
@@ -3,87 +3,110 @@ const presence = new Presence({
-		presenceData.state = document.querySelector(".post-title h1").textContent;
-		presenceData.smallImageKey = "view";
-		presenceData.buttons = [
-			{
-				label: "Visit Comic Page",
-				url: origin + pathname,
-			},
-		];
-	} else if (
-		/^\/series\/[0-9a-z-]+\/+(chapter|ch)-[0-9]+\/?$/i.test(pathname)
-	) {
-		let progress =
-			(document.documentElement.scrollTop /
-				(document.querySelector(".read-container").scrollHeight -
-					window.innerHeight)) *
-			100;
-		progress = Math.ceil(progress) > 100 ? 100 : Math.ceil(progress);
+	switch (pathSplit[0] ?? "") {
+		case "": {
+			const searchInput = document.querySelector<HTMLInputElement>("#search");
+			if (searchInput === document.activeElement) {
+				presenceData.details = "Searching";
+				presenceData.state = searchInput.value;
+				presenceData.smallImageKey = Assets.Search;
+			} else presenceData.details = "Viewing Home Page";
+			break;
+		}
+		case "comics":
+		case "novels": {
+			const captitalized = capitalize(pathSplit[0]);
+			if (pathSplit[1]) {
+				if (pathSplit[2] === "chapters") {
+					let progress =
+						(document.documentElement.scrollTop /
+							(document.querySelector(
+								pathSplit[0] === "comics"
+									? "main > div:not([id*='Ads'])"
+									: "article"
+							).scrollHeight -
+								window.innerHeight)) *
+						100;
+					progress = Math.ceil(progress) > 100 ? 100 : Math.ceil(progress);
 
-		presenceData.details = document.querySelector(
-			"ol.breadcrumb li:nth-child(3)"
-		).textContent;
-		presenceData.state = `📖 ${
-			document.querySelector("ol.breadcrumb li:nth-child(4)").textContent
-		} 🔸 ${progress}%`;
-		presenceData.smallImageKey = "read";
-		presenceData.buttons = [
-			{
-				label: "Visit Comic Page",
-				url:
-					origin +
-					document.querySelector<HTMLAnchorElement>(
-						"ol.breadcrumb li:nth-child(3) a"
-					).href,
-			},
-			{
-				label: "Visit Chapter",
-				url: origin + pathname,
-			},
-		];
-	} else {
-		presenceData.details = "Browsing Reaper Scans";
-		presenceData.state = document.title;
+					presenceData.details = `Reading ${document
+						.querySelector<HTMLParagraphElement>("main p")
+						.textContent.trim()}`;
+					presenceData.state = `${document
+						.querySelector<HTMLDivElement>("nav > div:nth-child(2)")
+						.textContent.trim()} - ${progress}%`;
+					presenceData.smallImageKey = Assets.Read;
+					presenceData.buttons = [
+						{
+							label: `Visit ${captitalized.slice(
+								captitalized.length - 1
+							)} Page`,
+							url: document.querySelector<HTMLAnchorElement>("h2 + div > a")
+								.href,
+						},
+						{
+							label: "Visit Chapter",
+							url: href,
+						},
+					];
+				} else {
+					presenceData.details = `Viewing ${captitalized} Page`;
+					presenceData.state = document.querySelector("h1").textContent;
+					presenceData.smallImageKey = Assets.View;
+					presenceData.buttons = [
+						{
+							label: `Visit ${captitalized} Page`,
```

Comment:
Labels sometimes have issues with displaying dynamic data, preferable that dynamic buttons aren't used.

Reply:
This code prevents the need for an if statement checking the path name. It can only have 2 possible values: "Novel" or "Chapter"